scripts/config-parser.py: prototype, example

filipleple opened the pull request.

filipleple commented:
```
 λ ./defconfig_parser.py config.input output.rob --check-unmapped
The following options are not included in the lib/mappings.json file:
  - CONFIG_PRERAM_CBMEM_CONSOLE_SIZE

You could extend the lib/mappings.json file to include these options.
```

mkopec commented on scripts/lib/mappings.json:
```diff
@@ -0,0 +1,23 @@
+    "CONFIG_EDK2_IPXE_OPTION_NAME": "IPXE_BOOT_ENTRY",
+    "CONFIG_TPM_MEASURED_BOOT": "SERIAL_BOOT_MEASURE",
+    "CONFIG_MAINBOARD_SMBIOS_PRODUCT_NAME": "DMIDECODE_PRODUCT_NAME",
+    "CONFIG_SYSTEM_TYPE_LAPTOP": "OPTIONS_LIB"
```
maybe CONFIG_EDK2_SERIAL_SUPPORT would be a better match here?

filipleple replied:
Yes, agreed, [changed here](https://github.com/Dasharo/open-source-firmware-validation/pull/392/commits/3c56ce6b6115def3dc0cd03478686b2c56445274#diff-63d9f43eb53de310c84792f587f9e3d6d867882fe3f7a4ec78e13dabf541a488R9)

mkopec replied:
whoops, sorry, with `CONFIG_EDK2_SERIAL_SUPPORT` it doesn't detect laptops properly...

Please revert :)

mkopec commented:
Looks pretty good. @filipleple I noticed that if I pass the full .config file from coreboot I get way more options.

```
./scripts/defconfig_parser.py -m scripts/lib/mappings.json ../coreboot/.config test                                                                                         2 defconfig-parser!?
m@phaaze~/D/D/osfv> cat test                                                                                                                                                                      defconfig-parser!?
${VERIFIED_BOOT_SUPPORT}=                         y
${DMIDECODE_PRODUCT_NAME}=                        V5xTNC_TND_TNE
${SERIAL_BOOT_MEASURE}=                           y
${DEFAULT_POWER_STATE_AFTER_FAIL}=                Powered Off
${OPTIONS_LIB}=                                   dcu
${TPM_SUPPORT}=                                   y
${SECURE_BOOT_SUPPORT}=                           y
${IPXE_BOOT_ENTRY}=                               iPXE Network Boot
```

vs 

```
m@phaaze~/D/D/osfv> ./scripts/defconfig_parser.py -m scripts/lib/mappings.json ../coreboot/defconfig test                                                                                         defconfig-parser!?
m@phaaze~/D/D/osfv> cat test                                                                                                                                                                      defconfig-parser!?
${VERIFIED_BOOT_SUPPORT}=                y
${SERIAL_BOOT_MEASURE}=                  y
```

is there any reason to pass the defconfig and not the full config?

By the way we can extract the full config from a coreboot binary. This could be scripted so that we only need to pass a complete binary (which we already need to have when running a test suite)  - but maybe in a future version.

mkopec commented:
Also please add some instructions on how to use this to the README

mkopec commented on scripts/lib/mappings.json:
```diff
@@ -0,0 +1,23 @@
+    "CONFIG_TPM": "TPM_SUPPORT",
+    "CONFIG_EDK2_SECURE_BOOT_SUPPORT": "SECURE_BOOT_SUPPORT",
+    "CONFIG_EDK2_IPXE_OPTION_NAME": "IPXE_BOOT_ENTRY",
+    "CONFIG_TPM_MEASURED_BOOT": "SERIAL_BOOT_MEASURE",
```
I'm pretty sure this one is just about measuring boot time with cbmem, nothing to do with TPM

filipleple replied:
Right, [removed](https://github.com/Dasharo/open-source-firmware-validation/pull/392/commits/3c56ce6b6115def3dc0cd03478686b2c56445274#diff-63d9f43eb53de310c84792f587f9e3d6d867882fe3f7a4ec78e13dabf541a488L8)

filipleple commented:
@mkopec I've added the docs, removed the example output file and renamed the script from defconfig- to config- since yeah, a full config should work better, no reason to restrict usage to the **def**configs
